Incorrect behavior with create() after destroy()

# Required information
 * Distribution: Debian
 * Distribution version: Stretch
 * python version: 3.5.3
 * The output of
   * `lxc-start --version`: 2.0.7
   * `uname -a`: Linux  4.9.0-3-amd64 lxc/lxc#1 SMP Debian 4.9.30-2+deb9u2 (2017-06-26) x86_64 GNU/Linux

# Issue description
After instantiating, creating, and destroying a linux container, calling create without re-instantiating returns false rather than recreating the container

# Steps to reproduce
1. test = lxc.Container("test")
2. test.create(template="debian")
3. test.destroy()
4. test.create(template="debian")

the second call simply returns false

Thanks for assigning it @stgraber! 

I've been a little lost trying to think of the solution. 

- The create and destroy functions in the lxc.c file in this python repo are eventually making a call to functions in `lxc/lxccontainer.h`
- I'm guessing that the fix is actually in the `lxc/lxc` repo in [`lxccontainer.c` ](https://github.com/lxc/lxc/blob/master/src/lxc/lxccontainer.c). 
- Moreover, I'm having a hard time trying to set up things to run. I had a file structure as follows:

- lxc
-- lxc 
-- python3-lxc

However, when I go to python3-lxc and run `python3 setup.py build`, I get an error saying that `lxc.c:27:10: fatal error: 'lxc/lxccontainer.h' file not found`

Could you please give me some direction here? Or do you think this issue is more involved than we think it is and would require a lot of context?

--- a/lxc/__init__.py
+++ b/lxc/__init__.py
@@ -217,6 +217,7 @@ class Container(_lxc.Container):
             "args" (optional) is a tuple of arguments to pass to the
             template. It can also be provided as a dict.
         """
+
         if isinstance(args, dict):
             tmp_args = []
             for item in args.items():
@@ -230,7 +231,22 @@ class Container(_lxc.Container):
         template_args['args'] = tuple(args)
         if bdevtype:
             template_args['bdevtype'] = bdevtype
+        if not self.defined:
+           self.save_config()
         return _lxc.Container.create(self, **template_args)
+
+    def destroy(self):
+        """
+            Deletes the container and clears its config
+        """
+
+        if not _lxc.Container.destroy(self):
+            return False
+        
+        # Clear the configuration to match destroyed container state.
+        self.clear_config()
+
+        return True
 
     def clone(self, newname, config_path=None, flags=0, bdevtype=None,
               bdevdata=None, newsize=0, hookargs=()):

